ev3dev2/motor.py

```diff
@@ -102,17 +117,27 @@ class SpeedNativeUnits(SpeedValue):
     """
 
     def __init__(self, native_counts):
-        self.native_counts = native_counts
+        self.native_counts = round(native_counts)
     
     def __str__(self):
         return str(self.native_counts) + " counts/sec"
 
+    def __mul__(self, other):
+
+        if isinstance(other, SpeedNativeUnits):
+            return SpeedNativeUnits(self.native_counts * other.native_counts)
+        elif isinstance(other, (float, int)):
+            return SpeedNativeUnits(self.native_counts * other)
+        else:
+            raise Exception("{} can only be multiplied by a SpeedNativeUnits, int, or float".format(self))
+
```

Comment:
This block is duplicated basically unchanged below. Can we move it to the base class by changing the first `if` to
```
if isinstance(other, self.__class__):
```
?

Reply:
The problem is each child class has its own unique variable that must be multiplied.